### PR TITLE
mqtt: add mqtt frames

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://suricata.readthedocs.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/5731
 
Draft PR to gather feedback

Describe changes:
- Registers `PDU`, `Header`, and `Data` frames for MQTT

suricata-verify-pr: 1064

### CODE REVIEW

File: rust/src/mqtt/mqtt.rs [hunk 1]
```diff
@@ -455,6 +467,8 @@ impl MQTTState {
             SCLogDebug!("request: handling {}", current.len());
             match parse_message(current, self.protocol_version, self.max_msg_len) {
                 Ok((rem, msg)) => {
+
+                    self.mqtt_frames(flow, &stream_slice, &msg);
```
Comment (hsadia538) [inline rust/src/mqtt/mqtt.rs:471]: Is this the right place to register header and data frame?

File: rust/src/mqtt/mqtt.rs [hunk 2]
```diff
@@ -589,6 +608,17 @@ impl MQTTState {
...
     }
+
+fn mqtt_frames(&mut self,flow: *const Flow, stream_slice: &StreamSlice, input: &MQTTMessage)  {
+        let hdr = stream_slice.as_slice();
+            //MQTT payload has a fixed header of 2 bytes
+        let _mqtt_ts_hdr = Frame::new(flow, stream_slice, hdr,2 , MQTTFrameType::Header as u8);
+            SCLogDebug!("nfs4_ts_hdr Frame {:?}", _mqtt_ts_hdr);
+        let data= &hdr[1..input.header.remaining_length as usize];
+        let _mqtt_ts_data = Frame::new(flow, stream_slice, data, input.header.remaining_length as i64, MQTTFrameType::Data as u8);
+            SCLogDebug!("nfs4_ts_ops Frame {:?}", _mqtt_ts_data);
```
Comment (hsadia538) [inline rust/src/mqtt/mqtt.rs:619]: This is still a WIP, I just wanted to get feedback if I am going in the right direction or not?

File: rust/src/mqtt/mqtt.rs [hunk 3]
```diff
@@ -497,12 +511,16 @@ impl MQTTState {
...
-    fn parse_response(&mut self, input: &[u8]) -> AppLayerResult {
+    fn parse_response(&mut self, flow: *const Flow,stream_slice: StreamSlice) -> AppLayerResult {
+        let input = stream_slice.as_slice();
         let mut current = input;
+
         if input.is_empty() {
             return AppLayerResult::ok();
         }
 
+        let _pdu = Frame::new(flow, &stream_slice, input, input.len() as i64, MQTTFrameType::Pdu as u8);
```
Comment (hsadia538) [inline rust/src/mqtt/mqtt.rs:522]: The suricata verify tests for PDU frame don't PASS as well.

Comment (jufajardini) [reply rust/src/mqtt/mqtt.rs:522]: Can you share that test, too? It's ok to just push it to the draft PR you've submitted :)

Comment (hsadia538) [reply rust/src/mqtt/mqtt.rs:522]: I have submitted the draft PR for suricata verify tests as well.

Comment (jufajardini) [reply rust/src/mqtt/mqtt.rs:522]: I saw it, but you've mentioned tests for pdu frames here, while in the SV PR I could only find a test for the header, that's why I thought you may have other tests that had not been committed there...

Comment (hsadia538) [reply rust/src/mqtt/mqtt.rs:522]: Oh Ok. I will push that too.

Comment (hsadia538) [reply rust/src/mqtt/mqtt.rs:522]: done

**Comment by hsadia538** [comment]
Followed by #8382

**Pull request closed without merging**